hash-db/src/lib.rs

```diff
@@ -20,24 +20,32 @@
 use std::fmt::Debug;
 #[cfg(feature = "std")]
 use std::hash;
+#[cfg(not(feature = "std"))]
+use core::hash;
+
 #[cfg(feature = "std")]
 pub trait MaybeDebug: Debug {}
 #[cfg(feature = "std")]
 impl<T: Debug> MaybeDebug for T {}
-
-#[cfg(not(feature = "std"))]
-use core::hash;
 #[cfg(not(feature = "std"))]
 pub trait MaybeDebug {}
 #[cfg(not(feature = "std"))]
 impl<T> MaybeDebug for T {}
 
 
-/// Empty prefix constant.
-pub static EMPTY_PREFIX: Prefix<'static> = &[];
+/// An empty prefix constant. Mainly use for comparison
+/// and for root nodes.
+pub static EMPTY_PREFIX: Prefix<'static> = (&[], (0,0));
 
-/// Prefix byte information for avoding rc.
-pub type Prefix<'a> = &'a[u8];
+/// The Prefix of a trie node. Prefix is nibble path up to
+/// the node in the trie.
+/// For a value node it is the value key without the Partial
+/// bytes (encoded in the node).
```

Comment:
> For a value node it is the value key without the Partial bytes (encoded in the node).

So does this mean that the value node key has no prefix? Is it possible that two value nodes in different parts of the tree have the same partial key and therefore same hash?

Reply:
Same partial key and same hash is a possibility yes, but at that point either the prefix are different or the key are the same (key = prefix ++ partial).
Does it mean that the value node key has no prefix? Ah ok, I will rephrase that.